nservicebus/scalability-and-ha/sender-side-distribution.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Scaling out with sender-side distribution
+summary: How to scale out with sender-side distribution when using the MSMQ transport.
+tags:
+- Scale Out
+- Routing
+- MSMQ
+---
+
+NServiceBus endpoints using the MSMQ transport are unable to use the competing consumers pattern to scale out by adding additional worker instances. Sender-side distribution is a method of scaling out an NServiceBus endpoint using the MSMQ transport, without relying on a centralized [distributor](distributor/) assigning messages to available workers.
+
+NOTE: Sender-side distribution is available in NServiceBus Versions 6 and above.
+
+When using sender-side distribution:
+
+* Multiple endpoint instances (deployed to different servers) are capable of processing a message that requires scaled-out processing.
+* A client sending a message is aware of all the endpoint instances that can process the message.
+* The client sends the message to a worker endpoint instance based on round-robin distribution, or a custom distribution strategy.
+
+Using sender-side distribution requires two parts. The first part maps message types to logical endpoints, and occurs in code. The second part maps logical endpoints to physical endpoint instances running on a specific machine.
+
+
+## Mapping logical endpoints
+
+To map message types to logical endpoints, use the following config:
+
+snippet:Routing-FileBased-Config
+
+**TODO: Consider splitting the snippet above into 2 snippets, splitting endpoint mappings from file-based instance mapping.**
+
+This creates mappings specifying that the `AcceptOrder` command is handled by the **Sales** endpoint, while the `SendOrder` command is handled by the **Shipping** endpoint.
+
+Meanwhile, the logical-to-physical mappings will be configured in the Routes.xml file, as this information is an operational concern that must be changed for deployment to multiple machines.
```

Comment:
I think the intro does that? This is the only link I found and I'm not sure exactly what text here to link to it: http://docs.particular.net/nservicebus/endpoints/